### ISSUE TITLE
PostgreSQL ne(boolean) should be rendered as is not instead of <>

### ISSUE THREAD

**Issue by axelfontaine** [jOOQ/jOOQ#4153]
With jOOQ 3.5.3, for PostgreSQL, if you have a nullable boolean column and you want to check whether it doesn't contain one of the three values (true, false, null), the most compact syntax is like this ```TBL.BOOL_COL.ne(false)``` which gets rendered incorrectly as ```"tbl"."my_col" <> false``` whereas it really should be ```"tbl"."my_col" is not false```

**Comment by axelfontaine** [comment]
Ideally there could also be convenience methods like isNotFalse() and isNotTrue(). This would make the API more symmetric and comparable to what is being offered for null (isNull() + isNotNull())

**Comment by lukaseder** [comment]
This is awkward. We have `Field.isTrue()` and `Field.isFalse()` methods that we've added before reading the SQL standard (or PostgreSQL feature lists). They're implemented completely different from the standard way of interpreting booleans.

I don't agree with changin the behaviour of `<>`. That's too clever and doesn't map 1:1 with SQL. It's perfectly valid for a query to check for `col <> false` and return `null` / `unknown` in case when `col is null`. Perhaps the user wants that behaviour.

I guess the right way to go forward is to wait for jOOQ 4.0 when we can break backwards compatibility for these cases, and implement the SQL standard then.

**Comment by katzyn** [comment]
BTW, there are three boolean tests in the Standard; the third one is `IS [ NOT ] UNKNOWN`.

**Comment by lukaseder** [comment]
Yes, thanks. The linking issue (#9722) mentioned `UNKNOWN` as well. I've fixed the title.

**Comment by joseph-pachod** [comment]
To ease reading, solutions proposed in #17462:
- Easy yet still misleading: add some javadoc to the methods
- Deprecate `isTrue`/`isFalse` in favor of new methods `equalTrue`/`equalFalse` to explicit the generated SQL
